<launch>
  <arg name="viewer" default="false" /> <!-- Must be globally qualified -->
  <arg name="manager" default="manager" /> <!-- Must be globally qualified -->
  <arg name="respawn" default="false" />
  <!-- TODO Arguments for debayer, interpolation methods? -->

  <arg     if="$(arg respawn)" name="bond" value="" />
  <arg unless="$(arg respawn)" name="bond" value="--no-bond" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" output="screen" />

  <!-- arguments for camera param uri -->
  <arg name="camera_info_file_right"
       default="$(find ps4eye)/camera_info/right.yaml" />
  <arg name="camera_info_file_left"
       default="$(find ps4eye)/camera_info/left.yaml" />
  <arg name="camera_info_url_default" default="file://$(find ps4eye)/camera_info/default.yaml" />
  <arg name="camera_info_url_right" default="file://$(arg camera_info_file_right)" />
  <arg name="camera_info_url_left" default="file://$(arg camera_info_file_left)" />

  <!-- arguments for tf coords -->
  <arg name="parent_frame" default="/map" />
  <arg name="parent_transform" default="0 0 1 -1.57 0 -1.57" />

  <!-- This launchfile should bring up a node that broadcasts a ros image
       transport on /webcam/image_raw -->
  <arg name="load_driver" default="true" />
  <arg name="DEVICE" default="/dev/video1"/>
  <!-- The GStreamer framerate needs to be an integral fraction -->
  <arg name="FPS" default="60/1"/>
  <!-- Available resolution should be 3448x808, 1748x408,1744x408,898x200,896x200 -->
  <arg name="width" default="1748"/>
  <arg name="height" default="408"/>
  <arg name="PUBLISH_FRAME" default="false"/>
  <node name="gscam_driver" pkg="nodelet" type="nodelet"
        args="load gscam/GSCamNodelet $(arg manager)"
        output="screen"
        if="$(arg load_driver)">
    <param name="camera_name" value="default"/>
    <param name="gscam_config" value="v4l2src device=$(arg DEVICE) ! video/x-raw-yuv,framerate=$(arg FPS),width=$(arg width),height=$(arg height) ! ffmpegcolorspace"/>
    <param name="frame_id" value="/ps4eye_frame"/>
    <param name="sync_sink" value="true"/>
    <param name="camera_info_url" value="$(arg camera_info_url_default)" />
  </node>

  <node name="ps4eye_proc" pkg="nodelet" type="nodelet"
        args="load ps4eye_tegra/ps4eye_proc $(arg manager)"
        output="screen">
    <param name="left_file_name"  value="$(arg camera_info_file_left)"  />
    <param name="right_file_name" value="$(arg camera_info_file_right)" />
  </node>

  <group ns="stereo">
    <node pkg="nodelet" type="nodelet" name="point_cloud2"
          args="load stereo_image_proc/point_cloud2 /$(arg manager)" />
  </group>
</launch>
